trunner/test_runner.py

```diff
@@ -156,6 +156,8 @@ def run_tests(self, tests: Sequence[TestOptions]):
         """
 
         fail, skip = 0, 0
+        # Reboot is required for the initial test.
```

Comment:
and previously wasn't? Seems like it's being taken care of by sub-clause (3) below

Reply:
Thanks for suggestions. It was done previously, because `should_reboot` was set True by default. Reboot means also spawning an emulator process for emu targets. So we always want to reboot before starting a test campaign - for emu targets the mentioned reason, for hw targets - reboot needed after flashing.

As we can see in CI, there was always reboot before a first test and we want to leave this approach.